### PR TITLE
Dynamic Loading Screen

### PR DESCRIPTION
**Pull request by waltonzt**
Hello, I wanted to take a shot at #4429.

Please review and let me know what you think.

### CODE REVIEW

**Comment by VIKTORVAV99** [comment]
The spinner looks good I suppose but I had imagined something more along these lines:
Spin-1s-200px
But I'll leave that up to the team to decide.

Besides that the it would be nice if we had a loading icon (just the icon no text) as well and that they where both was minified. If you need any help with the minification let me know.

**Comment by Kongkille** [comment]
This looks great, I will just await review from our designer @rasmusjeppsen!

**Comment by waltonzt** [comment]
@VIKTORVAV99 I can take a shot at creating something like that. Are you still thinking it moves in the same path as the one I provided?

Also what method would you recommend to minify?

**Comment by VIKTORVAV99** [comment]
> @VIKTORVAV99 I can take a shot at creating something like that. Are you still thinking it moves in the same path as the one I provided?
> 
> Also what method would you recommend to minify?

Either spinning in the path you created or just fading in and out would work pretty well I think.

For minifying it I would run it through svgo (online [interface](https://jakearchibald.github.io/svgomg/ ) here) but you need to be really careful with the options as a lot of transforms are enabled by default.

**Comment by waltonzt** [comment]
@VIKTORVAV99 What about these?

electricitymap-loading-icon
electricitymap-loading

**Comment by VIKTORVAV99** [comment]
> @VIKTORVAV99 What about these?
> 
> electricitymap-loading-icon
> electricitymap-loading
> 
> 

Looks great! 🎉

Thanks for making these!

**Comment by waltonzt** [comment]
@VIKTORVAV99 Awesome! I think I figured out the minify tool you shared so I will commit those changes.

**Comment by rasmusjeppsen** [comment]
> This looks great, I will just await review from our designer @rasmusjeppsen!

@waltonzt I have nothing to add other than these are great! Thank you for creating these, Zack!

**Comment by VIKTORVAV99** [comment]
@Kongkille Do you know if there is any other places we should use this "loading logo" or loading icon?

Otherwise I think this is ready to merge.

**Review by Kongkille — APPROVED** [review]
Sorry for the late review!

I think this is all the places at the moment, otherwise, we can just add it.

File: web/src/components/loadingplaceholder.jsx
Comment (VIKTORVAV99) [inline web/src/components/loadingplaceholder.jsx:5]:
```suggestion
  background: transparent url(${resolvePath('images/electricitymap-loading-icon.svg')}) no-repeat center center;
  background-size: 10rem;
```
Are the background-size changes needed?

**Review by VIKTORVAV99 — CHANGES_REQUESTED** [review]
Just noticed this isn't using the icon version.

File: web/src/components/loadingplaceholder.jsx
Comment (Kongkille) [reply web/src/components/loadingplaceholder.jsx]: No those would be breaking, I've fixed it another commit

Comment (Kongkille) [reply web/src/components/loadingplaceholder.jsx]: image

**Review by VIKTORVAV99 — APPROVED** [review]

**Pull request merged**
